src/Storage/DbalNestedSet.php

```diff
@@ -349,7 +349,13 @@ protected function moveSubTreeToPosition($newLeftPosition, Node $node) {
 
       // Calculate depth difference.
       $newNode = $this->getNodeAtPosition($newLeftPosition);
-      $depthDiff = $newNode->getDepth() - $node->getDepth();
+      if (!$newNode) {
+        // No other children at this position, new depth = 1.
+        $depthDiff = 1 - $node->getDepth();
```

Comment:
Not sure about this, perhaps its better to get the previous node?
The issue is when you have parent at end of tree, there is no-one at the next position